c10/util/BFloat16.h

```diff
@@ -6,44 +6,51 @@
 #include <c10/macros/Macros.h>
 #include <cmath>
 #include <cstring>
-#include <iostream>
 
 namespace c10 {
 
-struct alignas(2) BFloat16 {
-  uint16_t val_;
+namespace detail {
+  inline C10_HOST_DEVICE bool isSmallEndian() {
+    int num = 1;
+    return *(char *)&num == 1;
+  }
 
-  struct from_bits_t {};
-  static constexpr from_bits_t from_bits() {
-    return from_bits_t();
+  inline C10_HOST_DEVICE float f32_from_bits(uint16_t src) {
+    float res = 0;
+    uint16_t* tmp = reinterpret_cast<uint16_t*>(&res);
```

Comment:
Could we have a comment here explaining why we are violating strict aliasing? :)